File: frontend/lib/features/payment_method/domain/entities/payment_method.dart
import 'package:equatable/equatable.dart';

class PaymentMethod extends Equatable {
  final String id;
  final String name;
  final String type;
  final int? settlementDay;
  final int? closingDay;
  final bool isActive;
  final bool isDefault;
  final int displayOrder;
  final int? balance;
  final String? linkedBankId;
  final String? linkedBankName;
  final DateTime createdAt;

  const PaymentMethod({
    required this.id,
    required this.name,
    required this.type,
    this.settlementDay,
    this.closingDay,
    required this.isActive,
    required this.isDefault,
    required this.displayOrder,
    this.balance,
    this.linkedBankId,
    this.linkedBankName,
    required this.createdAt,
  });

  bool get isCash => type == 'CASH';
  bool get isDebit => type == 'DEBIT';
  bool get isCredit => type == 'CREDIT';
  bool get isBank => type == 'BANK';

  PaymentMethod copyWith({
    String? id,
    String? name,
    String? type,
    int? settlementDay,
    int? closingDay,
    bool? isActive,
    bool? isDefault,
    int? displayOrder,
    int? balance,
    String? linkedBankId,
    String? linkedBankName,
    DateTime? createdAt,
  }) {
    return PaymentMethod(
      id: id ?? this.id,
      name: name ?? this.name,
      type: type ?? this.type,
      settlementDay: settlementDay ?? this.settlementDay,
      closingDay: closingDay ?? this.closingDay,
      isActive: isActive ?? this.isActive,
      isDefault: isDefault ?? this.isDefault,
      displayOrder: displayOrder ?? this.displayOrder,
      balance: balance ?? this.balance,
      linkedBankId: linkedBankId ?? this.linkedBankId,
      linkedBankName: linkedBankName ?? this.linkedBankName,
      createdAt: createdAt ?? this.createdAt,
    );
  }

  @override
  List<Object?> get props => [
        id,
        name,
        type,
        settlementDay,
        closingDay,
        isActive,
        isDefault,
        displayOrder,
        balance,
        linkedBankId,
        linkedBankName,
        createdAt,
      ];
}

File: backend/src/main/kotlin/com/budgetbook/paymentmethod/domain/PaymentMethod.kt
package com.budgetbook.paymentmethod.domain

import com.budgetbook.common.entity.BaseTimeEntity
import com.budgetbook.couple.domain.Couple
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "payment_methods")
class PaymentMethod(
    @Id
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "couple_id", nullable = false)
    val couple: Couple,

    @Column(nullable = false, length = 100)
    var name: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    val type: PaymentMethodType,

    @Column(name = "settlement_day")
    var settlementDay: Int? = null,

    @Column(name = "closing_day")
    var closingDay: Int? = null,

    @Column(name = "is_active", nullable = false)
    var isActive: Boolean = true,

    @Column(name = "is_default", nullable = false)
    val isDefault: Boolean = false,

    @Column(name = "display_order", nullable = false)
    var displayOrder: Int = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "linked_bank_id")
    var linkedBank: PaymentMethod? = null
) : BaseTimeEntity()

enum class PaymentMethodType { CASH, DEBIT, CREDIT, BANK }
